pkg/controller/nodelink/nodelink_controller_test.go

```diff
@@ -0,0 +1,495 @@
+package nodelink
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	mapiv1beta1 "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func init() {
+	if err := mapiv1beta1.AddToScheme(scheme.Scheme); err != nil {
+		klog.Fatal(err)
+	}
+}
+
+const (
+	ownerControllerKind = "MachineSet"
+	namespace           = "openshift-machine-api"
+)
+
+var (
+	knownDate          = metav1.Time{Time: time.Date(1985, 06, 03, 0, 0, 0, 0, time.Local)}
+	fakeNodeIndexer    = make(map[string]corev1.Node)
```

Comment:
It would be better if these were not globals. Create them in the tests that need them. Modify build() funcs to return a map. Otherwise it is not clear from Test to Test function whether they should/should not have any content.

Reply:
Done in newFakeReconciler